src/towncrier/test/test_settings.py

```diff
@@ -192,6 +194,22 @@ def test_towncrier_toml_preferred(self):
         config = load_config(project_dir)
         self.assertEqual(config.package, "a")
 
+    def test_load_no_config(self):
+        """
+        Calling the root CLI without an existing configuration file in the base directory,
+        will exit with code 1 and an informative message is sent to standard output.
+        """
+        temp = self.mktemp()
+        os.makedirs(temp)
```

Comment:
Do we need this?

Reply:
Yes, but it could use the `@with_isolated_runner` decorator found in `towncrier.test.helpers` instead to simplify things (and the runner can be invoked without `["--dir", temp]`)

Reply:
> (and the runner can be invoked without `["--dir", temp]`)

This doesn't seem to work? The runner looks in a subdirectory below `temp`, for example in `/tmp/tmpjsgwk7qd/towncrier.test.test_settings/TomlSettingsTests/test_load_no_config/t5o5_2b0/temp` instead of `/tmp/tmpjsgwk7qd`

Reply:
> (and the runner can be invoked without `["--dir", temp]`)

This doesn't seem to work? The runner looks in a subdirectory below `temp`, for example in `/tmp/tmpjsgwk7qd/towncrier.test.test_settings/TomlSettingsTests/test_load_no_config/t5o5_2b0/temp` instead of `/tmp/tmpjsgwk7qd`

Reply:
Are y'all sure about the mktemp? The docs of it claim that it creates the directory and that's what I'd expect!? The name is literally "make temp"?? 😵‍💫

Reply:
Ugh you're right https://docs.twistedmatrix.com/en/stable/api/twisted.trial.unittest.SynchronousTestCase.html#mktemp I was sure I checked this before. I knew the OG mktemp is deprecated because of race condition problems but I was under the wrong impression that trial did it right. :( Pls disregard. _mumble pytest mumble_